File: frontend/bookease_app/lib/models/slot.dart
class Slot {
  final String id;
  final String serviceId;
  final String serviceName;
  final String startTime;
  final String endTime;
  final int currentBookings;
  final int capacity;
  final bool isAvailable;
  final bool isActive;

  const Slot({
    required this.id,
    required this.serviceId,
    required this.serviceName,
    required this.startTime,
    required this.endTime,
    required this.currentBookings,
    required this.capacity,
    required this.isAvailable,
    required this.isActive,
  });

  factory Slot.fromJson(Map<String, dynamic> json) => Slot(
        id: json['id'] as String,
        serviceId: json['serviceId'] as String,
        serviceName: json['serviceName'] as String,
        startTime: json['startTime'] as String,
        endTime: json['endTime'] as String,
        currentBookings: json['currentBookings'] as int,
        capacity: json['capacity'] as int,
        isAvailable: json['isAvailable'] as bool,
        isActive: json['isActive'] as bool,
      );
}

File: backend/BookEase.API/Models/Slot.cs
namespace BookEase.API.Models;

public class Slot
{
    public Guid Id { get; set; }
    public Guid ServiceId { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public int CurrentBookings { get; set; }
    public bool IsActive { get; set; }

    // Navigation
    public Service Service { get; set; } = null!;
    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}
